src/vm.rs

```diff
@@ -70,18 +71,51 @@ impl PevmTxExecutionResult {
     }
 }
 
-// TODO: Rewrite as [Result]
-pub(crate) enum VmExecutionResult {
+pub(crate) enum VmExecutionError {
     Retry,
     FallbackToSequential,
-    ReadError {
-        blocking_tx_idx: TxIdx,
-    },
+    Blocking { blocking_tx_idx: TxIdx },
```

Comment:
Sorry, I should have been clearer. But we should also change this to `Blocking(TxIdx)`, so no more `blocking_tx_idx` for both error types.